### PR TITLE
Support UTF-32 conversion.

### PR DESCRIPTION
**Pull request by olonho**

### CODE REVIEW

**Comment by msink** [comment]
Maybe better as in C++: `u16str` and `u32str` always 16/32 bit, and `wcstr` depends on `sizeof(wchar_t)` ?

**Comment by msink** [comment]
BTW - if I want to compile some K/N library as shared lib for Windows/Linux/Mac, and export all Kotlin strings on Linux and Mac as utf8 and on Windows as utf16 (no conversions) - is it possible somehow?

File: Interop/Runtime/src/native/kotlin/kotlinx/cinterop/NativeMem.kt [hunk 1]
```diff
@@ -150,15 +150,45 @@ fun CPointer<UShortVar>.toKString(): String {
...
     while (index < length) {
-        bytes[index] = nativeBytes[index].toShort().toChar()
+        chars[index] = nativeBytes[index].toShort().toChar()
         ++index
     }
-    return String(bytes)
+    return String(chars)
 }
 
+fun CPointer<IntVar>.toKString(): String {
```
Comment (SvyatoslavScherbina) [inline Interop/Runtime/src/native/kotlin/kotlinx/cinterop/NativeMem.kt]: Shouldn't we mention UTF-32 explicitly in the name of this function?

Comment (olonho) [reply Interop/Runtime/src/native/kotlin/kotlinx/cinterop/NativeMem.kt]: How would you suggest to call it?

Comment (SvyatoslavScherbina) [reply Interop/Runtime/src/native/kotlin/kotlinx/cinterop/NativeMem.kt]: I don't know. Maybe `utf32ToKString`, `toKStringFromUTF32`, `decodeUtf32` or `decodeFromUtf32` etc.
The second variant appears to be more discoverable (considering existing `toKString` functions), but we may decide to cut discoverability down to be able to choose better name.

**Comment by olonho** [comment]
> BTW - if I want to compile some K/N library as shared lib for Windows/Linux/Mac, and export all Kotlin strings on Linux and Mac as utf8 and on Windows as utf16 (no conversions) - is it possible somehow?

Internally strings are in UTF-16, so it's unlikely possible.

File: Interop/Runtime/src/native/kotlin/kotlinx/cinterop/NativeMem.kt (hunk 1 above)
Comment (olonho) [reply Interop/Runtime/src/native/kotlin/kotlinx/cinterop/NativeMem.kt]: toKStringFromUtf32 looks OKish to me.

**Pull request merged**